nuitka/plugins/standard/PlaywrightPlugin.py

```diff
@@ -0,0 +1,168 @@
+#     Copyright 2024, Kay Hayen, mailto:kay.hayen@gmail.com find license text at end of file
+
+""" Plugin for Playwright.
+
+spell-checker: ignore Playwright
+"""
+
+from nuitka.Options import isStandaloneMode
+from nuitka.plugins.PluginBase import NuitkaPluginBase
+import os
+from nuitka.utils.Execution import NuitkaCalledProcessError
+
+
+class NuitkaPluginPlaywright(NuitkaPluginBase):
+    """This class represents the main logic of the plugin."""
+
+    plugin_name = "Playwright"
+    plugin_desc = "Required by 'playwright' package."
+
+    def __init__(self, include_browsers):
+        self.include_browsers = list(include_browsers)
+        self.installed_browsers = self.get_Installed_Playwright_Browsers()
+
+        if "all" in self.include_browsers:
+            self.include_browsers = self.installed_browsers
+        elif "ffmpeg" not in self.include_browsers and any(
+            browser.startswith("chrom") for browser in self.include_browsers
+        ):
+            for browser in self.installed_browsers:
+                if "ffmpeg" in self.installed_browsers[browser].name:
+                    self.include_browsers.append(browser)
+                    self.info(
+                        "Including 'ffmpeg' for chromium-based browser. It is required by playwright."
+                    )
+                    break
+
+        for browser in self.include_browsers:
+            if browser not in self.installed_browsers:
+                self.sysexit(
+                    f"Error, requested to include browser '{browser}' that was not found, the list of installed ones is '{', '.join(self.installed_browsers.keys())}'."
+                )
+
+    @staticmethod
+    def isAlwaysEnabled():
+        return True
+
+    @staticmethod
+    def isRelevant():
+        return isStandaloneMode()
+
+    @classmethod
+    def addPluginCommandLineOptions(cls, group):
+        group.add_option(
+            "--playwright-include-browser",
+            action="append",
+            dest="include_browsers",
+            default=[],
+            help="""\
+            Playwright browser to include. Can be specified multiple times. use "all" to include all installed browsers.
+            """,
+        )
+
+    def _getPlaywrightInfo(self):
+        "Determine the path of the playwright module."
+        try:
+            info = self.queryRuntimeInformationMultiple(
+                info_name="playwright_info",
+                setup_codes="""
+from playwright import __file__ as playwright_file
```

Comment:
Use `self.locateModule` instead of you only want a module location at compile time.